torch/csrc/jit/symbolic_script.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+// This file is temporary until native_functions.yaml and derivatives.yaml are merged.
+// Ideally this should all go into native_functions.yaml
+
+namespace torch { namespace jit {
+  const std::unordered_map<std::string, std::string> symbolic_scripts({
+    {"aten::mul(Tensor self, Tensor other) -> Tensor",
+R"(
```

Comment:
For broadcasting we probably want to express this as:
```
    return reduce_to(grad_output * other, self.size()), reduce_to(grad_output * self, other.size()
```
Where `reduce_to(value, size)` is the backward of broadcast. i.e. it reduces the dimensions of the tensors back to size of the original value through summation. We probably have this op implemented but I don't know its name offhand. A shape-aware optimization pass could then eliminate stuff. This overlaps a lot with the changes @apaszke did to make broadcasting work again, but I do not know enough of the details to say more.